### PR TITLE
Various http stream test improvements, doc update

### PR DESCRIPTION
**Pull request by martinpitt**
Splitting this out from PR #17631 as the /mock/info stuff might be useful for @jelly and @mvollmer to adjust their parts of the unit test for py vs. C bridge differences.

### CODE REVIEW

File: pkg/base1/test-echo.js [hunk 1]
```diff
@@ -79,41 +79,55 @@ QUnit.test("binary", function (assert) {
...
-    }
-
-    /* We send messages in this order, but they should echoed in numeric order */
-    before.send("1");
-    after.send("4");
-    after.send("5");
-    fence.send("2");
-    fence.send("3");
+    // This is implemented in the C bridge, but not in Python.
+    cockpit.http(test_server).get("/mock/info")
```
Comment (allisonkarlitskaya) [inline pkg/base1/test-echo.js:91]: Less chaining and arrows.  More `await`/`async`, please.

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]
Looks good except for one major issue.  I'm gonna start getting more and more annoying on this point, going forward... ;)

**Comment by martinpitt** [comment]
Oops, this also needs work -- `make pycheck` hangs. cockpit.http() on /mock/info can't work on the pybridge yet in this PR, so this is bogus. I'll drop this and re-put everything into #17631

File: pkg/base1/test-echo.js (hunk 1 above)
Comment (martinpitt) [reply pkg/base1/test-echo.js]: Done.

File: pkg/base1/test-echo.js [hunk 2]
```diff
@@ -79,10 +79,18 @@ QUnit.test("binary", function (assert) {
     channel.send(buffer);
 });
 
-// This is implemented in the C bridge, but not in Python.
-// Nobody uses it, so skip the test for now.
-QUnit.test.skip("fence", function (assert) {
+QUnit.test("fence", async assert => {
     const done = assert.async();
+
+    const response = await fetch(`http://${window.location.hostname}:${window.location.port}/mock/info`);
```
Comment (allisonkarlitskaya) [inline pkg/base1/test-echo.js:85]: Could we integrate that into `pkg/lib/qunit-tests.js` somehow?

**Review by allisonkarlitskaya — APPROVED** [review]
Looks good, but please see suggestion: if we want to use this more often then the current approach is a bit too anti-DRY for my tastes.

File: pkg/base1/test-echo.js (hunk 2 above)
Comment (martinpitt) [reply pkg/base1/test-echo.js:85]: Indeed in the http channel PR I [have a helper function](https://github.com/martinpitt/cockpit/commit/85f54d839bd22b36de7bfc4d725494a22401603e#diff-8f24a1fc445b04e36587c1207a30cec18e7973665db3543ef9e8854530758e95R18) already. Moving to qunit-tests.js sounds good, I'll experiment with that there, do a rebase, and centralize it.

Comment (martinpitt) [reply pkg/base1/test-echo.js:85]: Done in https://github.com/cockpit-project/cockpit/pull/17631/commits/ab40c130da35a517076e0adefafcccf607c158e0

**Pull request merged**